src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -541,7 +541,10 @@ public ICollection<string> GetWarningsToBeLoggedAsErrorsByProject(BuildEventCont
 
             int key = GetWarningsAsErrorOrMessageKey(context);
 
-            HashSet<string> warningsAsErrorsExcludingMessages = new HashSet<string>(_warningsAsErrorsByProject[key]);
+            if (!_warningsAsErrorsByProject.TryGetValue(key, out ISet<string> warningsAsErrorsExcludingMessages))
```

Comment:
I was going to suggest checking `!_warningsAsErrorsByProject.ContainsKey(key)`, but [containskey actually calls trygetvalue](https://source.dot.net/#System.Collections.Concurrent/System/Collections/Concurrent/ConcurrentDictionary.cs,93f90f07cf34d76c,references).